Sources/GenerateValFileTests/GenerateValFileTests.swift

```diff
@@ -0,0 +1,153 @@
+import ArgumentParser
+import Foundation
+
+/// A command-line tool that, given a list of .val files, generates XCTest cases for annotated .val
+/// files as part of our build process.
+@main
+struct GenerateValFileTests: ParsableCommand {
+
+  @Option(
+    name: [.customShort("o")],
+    help: ArgumentHelp("Write output to <file>.", valueName: "output-swift-file"),
+    transform: URL.init(fileURLWithPath:))
+  var outputURL: URL
+
+  @Option(
+    name: [.customShort("n")],
+    help: ArgumentHelp("Name of generated test case.", valueName: "XCTestCase name"))
+  var testCaseName: String
+
+  @Argument(
+    help: "Paths of annotated val source files to be tested.",
+    transform: URL.init(fileURLWithPath:)
+  )
+  var valSourceFiles: [URL]
+
+  /// Returns the Swift source of the test function for the Val at `source`, having `firstLine` as
+  /// its first line.
+  func swiftFunctionTesting(valAt source: URL, withFirstLine firstLine: Substring) throws -> String
+  {
+    let parsed = try firstLine.parsedAsFirstLineOfAnnotatedValFileTest()
+    let testID = source.deletingPathExtension().lastPathComponent.asSwiftIdentifier
+
+    return """
+
+        func test_\(parsed.methodName)_\(testID)() throws {
+          try \(parsed.methodName)(
+            \(String(reflecting: source.path)), expectSuccess: \(parsed.expectSuccess))
+        }
+
+      """
+  }
+
+  func run() throws {
+    var output =
+      """
+      import XCTest
+
+      final class \(testCaseName.asSwiftIdentifier): XCTestCase {
+
+      """
+
+    for f in valSourceFiles {
+      let firstLine = try String(contentsOf: f).split(separator: "\n", maxSplits: 1).first ?? ""
+      do {
+        output += try swiftFunctionTesting(valAt: f, withFirstLine: firstLine)
+      } catch let e as FirstLineError {
+        try! FileHandle.standardError.write(
+          contentsOf: Data("\(f.path):1: error: \(e.details)\n".utf8))
+        GenerateValFileTests.exit(withError: ExitCode(-1))
+      }
+    }
+
+    output += "\n}\n"
+
+    try output.write(to: outputURL, atomically: true, encoding: .utf8)
+
+  }
+}
+
+extension String {
+
+  /// Returns a valid Swift identifier made by replacing non-identifier characters in `self` with
+  /// underscores, and prefixing it with "X" if it happens to begin with a number.
+  fileprivate var asSwiftIdentifier: String {
+    let r = String(self.lazy.map { $0.isNumber || $0.isLetter ? $0 : "_" })
```

Comment:
```suggestion
    let r = String(self.map { $0.isNumber || $0.isLetter ? $0 : "_" })
```

`lazy.map` is almost always slower than an eager `map` for this kind of small closures. I ran benchmarks. The fastest approach is `reduce`, but it's slightly harder to read and the gain is too small to justify the extra complexity.

```suggestion
    let r = reduce(into: "", { (s, c) in s.append((c.isNumber || c.isLetter) ? c : "_") })
```

Reply:
It's a crime that an eager map is faster, but thanks.